### PR TITLE
retry failed search item

### PR DESCRIPTION
**Pull request by paraselene**
## Description

If the search worker failed to scan, it will retry for the failed search item next round

## Motivation and Context

Currently, If it failed in one search item, it won't retry for the same location again

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by ic4roswings** [comment]
@paraselene  Does it works with 0/0/0 or if a captcha is encountered ?

**Comment by neskk** [comment]
Nicely done ;)

**Comment by sclo012** [comment]
Is there any way to rebase it?

File: pogom/search.py [hunk 1]
```diff
@@ -561,12 +566,14 @@ def search_worker_thread(args, account_queue, account_failures, search_items_que
...
+                        if retry_failed_search_item is None:
+                            search_items_queue.task_done()
                         continue
 
                 # Too late?
                 if leaves and now() > (leaves - args.min_seconds_left):
-                    search_items_queue.task_done()
+                    if retry_failed_search_item is None:
+                        search_items_queue.task_done()
                     status['skip'] += 1
```
Comment (neskk) [inline pogom/search.py]: @kbtbc I think I had this problem before too, it really doesn't make sense to retry a "Too late" scan...
We need to ensure that a "Too late" item is not scheduled for a retry and if for some reason we're retrying an item that is "Too late" then we need to set `retry_failed_search_time =None`.

**Comment by sLoPPydrive** [comment]
Should be closed as #1585 is the cleaner solution that has additional checks whether a scan is really empty (-> no gyms, no nearby pokemon, no pokestops) or just does not have any spawns.

**Pull request closed without merging**